### ISSUE TITLE
[Bug]: Crash when turning on/off experiments if using languages which don't use dots.

### ISSUE THREAD

**Issue by hokonch** [mozilla-mobile/fenix#21148]
### Steps to reproduce

1. Change language to Chinese
2. Go to Data collection
3. Disable or enable Studies

### Expected behaviour

Experiments turned on/off successfully 

### Actual behaviour

Crash and can't  turn on/off experiments.

### Firefox release type

Firefox & Firefox Beta

### Firefox version

- Release: from v91.3.1 to v92.1.1
- Beta: from 92.0.0-beta.6 to 92.0.0-beta.8

### Device logs

----
f58b914f-fcdc-4b01-8637-2646ab9c8fc0
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
 * Sentry: https://sentry.prod.mozaws.net/operations/fenix-fennec/?query=ed2118949f3b48a2a35decebc8b1a616
 * Socorro: https://crash-stats.mozilla.org/report/index/bp-3611bc26-3362-4389-b50f-cecba0210904
----
~~~
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
	at java.util.Collections$SingletonList.get(Collections.java:4872)
	at org.mozilla.fenix.settings.DataChoicesFragment.access$getQuittingAppString(DataChoicesFragment.kt:2)
	at org.mozilla.fenix.settings.DataChoicesFragment$onCreatePreferences$$inlined$apply$lambda$1.onPreferenceChange(Extensions.kt:8)
	at androidx.preference.Preference.callChangeListener(Preference.java:1)
	at androidx.preference.TwoStatePreference.onClick(TwoStatePreference.java:2)
	at androidx.preference.Preference.performClick(Preference.java:3)
	at androidx.preference.SwitchPreference.performClick(SwitchPreference.java:1)
	at androidx.preference.Preference$1.onClick(Preference.java:1)
	at android.view.View.performClick(View.java:7448)
	at android.view.View.performClickInternal(View.java:7425)
	at android.view.View.access$3600(View.java:810)
	at android.view.View$PerformClick.run(View.java:28305)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7664)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
~~~

### Additional information

https://github.com/mozilla-mobile/fenix/pull/20947

https://github.com/mozilla-mobile/fenix/blob/40a797b6b31bf528b74df4d69b8eb582eeebaa55/app/src/main/java/org/mozilla/fenix/settings/DataChoicesFragment.kt#L103-L114

**Will definitely crash**
~~~
./app/src/main/res/values-bn/strings.xml:    <string name="toast_override_fxa_sync_server_done">Firefox Account/Sync সার্ভার সংশোধিত হয়েছে। পরিবর্তনগুলি প্রয়োগ করতে অ্যাপ্লিকেশন বন্ধ করুন…</string>
./app/src/main/res/values-hi-rIN/strings.xml:    <string name="toast_override_fxa_sync_server_done">Firefox Account/सिंक सर्वर संसोधित किया गया। परिवर्तन लागू करने के लिए एप्लीकेशन बंद हो रहा है…</string>
./app/src/main/res/values-hy-rAM/strings.xml:    <string name="toast_override_fxa_sync_server_done">Firefox հաշիվը / համաժամեցման սերվերը փոփոխված է: Հայտը դադարեցվում է՝ փոփոխությունները գործադրելու համար…</string>
./app/src/main/res/values-ja/strings.xml:    <string name="toast_override_fxa_sync_server_done">Firefox アカウント/Sync サーバーが変更されました。変更を適用するためにアプリケーションを終了しています…</string>
./app/src/main/res/values-kab/strings.xml:    <string name="toast_override_fxa_sync_server_done">Amiḍan Firefox/Aqeddac n umtawi ittwasnifel, Ffeɣ seg usnas akken ad iddu usnifel…</string>
./app/src/main/res/values-my/strings.xml:    <string name="toast_override_fxa_sync_server_done">Firefox Account/Sync server ကို ပြုပြင်ပြီး ဖြစ်သည်။ အပြောင်းအလဲများ လျှောက်ထားရန်လျှောက်လွှာ မှ ထွက်ပါ</string>
./app/src/main/res/values-pa-rIN/strings.xml:    <string name="toast_override_fxa_sync_server_done">Firefox ਖਾਤਾ/ਸਿੰਕ ਸਰਵਰ ਸੋਧਿਆ ਗਿਆ ਹੈ। ਤਬਦੀਲੀਆਂ ਲਾਗੂ ਕਰਨ ਲਈ ਐਪਲੀਕੇਸ਼ਨ ਤੋਂ ਬਾਹਰ ਜਾਓ…</string>
./app/src/main/res/values-sat/strings.xml:    <string name="toast_override_fxa_sync_server_done">Firefox ᱠᱷᱟᱛᱟ/ ᱥᱭᱝᱠ ᱠᱟᱹᱨᱣᱟᱹᱨ ᱵᱚᱫᱚᱞᱮᱱᱟ ᱾ ᱵᱚᱫᱚᱞ ᱠᱚ ᱫᱚᱦᱚ ᱞᱟᱹᱜᱤᱫ ᱮᱯᱯᱞᱤᱠᱮᱥᱚᱱ ᱵᱚᱸᱫᱚᱜ ᱠᱟᱱᱟ…</string>
./app/src/main/res/values-th/strings.xml:    <string name="toast_override_fxa_sync_server_done">เปลี่ยนแปลงเซิร์ฟเวอร์บัญชี Firefox/Sync แล้ว กำลังออกจากแอปพลิเคชันเพื่อนำการเปลี่ยนแปลงไปใช้…</string>
./app/src/main/res/values-ur/strings.xml:    <string name="toast_override_fxa_sync_server_done">Firefox Account/Sync سرور بدل دیا گیا۔ تبدیلیاں لاگو کرنے کے لیے ایپلیکیشن کو بند کیا جا رہا ہے…</string>
./app/src/main/res/values-zh-rCN/strings.xml:    <string name="toast_override_fxa_sync_server_done">已更改 Firefox 账户/同步服务器设置。退出应用程序以应用更改…</string>
./app/src/main/res/values-zh-rTW/strings.xml:    <string name="toast_override_fxa_sync_server_done">已更改 Firefox 帳號 / 同步伺服器設定。將結束應用程式讓變更生效…</string>
~~~

**Bad message**
~~~
./app/src/main/res/values-dsb/strings.xml:    <string name="toast_override_fxa_sync_server_done">Kontowy resp. synchronizěrowański serwer Firefox jo se změnił. Nałoženje se kóńcy, aby se změny nałožyli…</string>
./app/src/main/res/values-hsb/strings.xml:    <string name="toast_override_fxa_sync_server_done">Kontowy resp. synchronizowanski serwer Firefox je so změnił. Nałoženje so kónči, zo bychu so změny nałožili…</string>
~~~

https://user-images.githubusercontent.com/54430032/132104259-2fa0a46a-629b-4cb2-9956-5499c56b6cd3.mp4

**Comment by Amejia481** [comment]
We put a patch https://github.com/mozilla-mobile/fenix/pull/21233 it should be on the new release version, for Beta users the issue will be fixed when they update to version `93.0.0` or above.

**Comment by sflorean** [comment]
I was able to reproduce the issue on the affected builds with Pixel 3 (Android 12) and I can confirm that the issue is fixed on Nightly 94, and Beta 93 versions, following the steps from the description.
